account.rs/src/poseidon.ts

```diff
@@ -0,0 +1,45 @@
+import { poseidon as wasmPoseidon } from "light-wasm";
+import { featureFlags } from "./config";
+import { BN } from "@coral-xyz/anchor";
+const circomlibjs = require("circomlibjs");
+
+export class Poseidon {
+  private static instance: Poseidon;
+
+  private circomPoseidon: any | undefined;
+  private constructor() {}
+
+  public static async getInstance(): Promise<Poseidon> {
+    if (!Poseidon.instance) {
+      Poseidon.instance = new Poseidon();
+    }
+    if (!featureFlags.wasmPoseidon) {
+      Poseidon.instance.circomPoseidon = await circomlibjs.buildPoseidonOpt();
```

Comment:
We keep it here just in case we want to roll back to circom poseidon. 
And even if we eventually decide to get rid of circomlibjs dependency completely, it would be much easier than now, because in this case we only have to remove a couple lines of code.